### PR TITLE
feat: Support Q4_0

### PR DESCRIPTION
**Pull request by PsiACE**
- add support for q4_0
- lack of testing

### CODE REVIEW

File: crabml-core/src/backends/cpu/buf/buf_q4_0.rs [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        bs.push(BlockQ4_0 {
+            d: f16::from_f32(d),
+            qs,
+        });
+    }
+
+    bs
+}
+
+pub fn vec_dot_q4_0_q4_0_naive(abs: &[BlockQ4_0], bbs: &[BlockQ4_0]) -> f32 {
```
Comment (PsiACE) [inline crabml-core/src/backends/cpu/buf/buf_q4_0.rs:132]: There is an issue here, we need to support vec_dot for Q4_0 and Q8_0.

**Comment by PsiACE** [comment]
I have re-quantified a Q4_0 tinyllamas-stories model. Now I just need to fix vec_dot.

**Review by flaneur2020 — APPROVED** [review]
LGTM, thank you for your contribution! 🤩

**Review by Xuanwo — APPROVED** [review]
Perfect! Let's move.

**Pull request merged**